### ISSUE TITLE
Handle different directory file separators for different Operating Systems 

### ISSUE THREAD

**Issue by jayasuryat** [jayasuryat/mendable#35]
So turns out, different `Operating Systems` use different `directory file separators` (the character between two directories in the file/directory path, for example `\` in `C:\Users\me\Desktop`). Read more [here](https://en.wikipedia.org/wiki/Path_%28computing%29#Representations_of_paths_by_operating_system_and_shell).

I just came across this while running tests on `Windows` `Operating System`, some of the tests were failing which were working fine on `CI` which was using `Ubuntu`. For example [this one](https://github.com/jayasuryat/mendable/blob/fed58939f134752207dfb6864ca615d4845e15b5/mendable/src/test/java/com/jayasuryat/mendable/MendableReportGeneratorTest.kt#L110), it was failing as a [file path was being constructed](https://github.com/jayasuryat/mendable/blob/fed58939f134752207dfb6864ca615d4845e15b5/mendable/src/test/java/com/jayasuryat/mendable/MendableReportGeneratorTest.kt#L130) with an improper directory file separator for the platform.

Looks like `java.io.File` has a `static property` `File.separator` which returns the system-dependent default name-separator character which could be used to solve this problem.

---

### TLDR

Instead of using a hardcoded character for `directory file separators` use `java.io.File.separator`

**Comment by jayasuryat** [comment]
Only [this](https://github.com/jayasuryat/mendable/blob/6d19100720b017b5c77092acd584b3aa118a00a4/mendable/src/test/java/com/jayasuryat/mendable/MendableReportGeneratorTest.kt#L130) and [this](https://github.com/jayasuryat/mendable/blob/6d19100720b017b5c77092acd584b3aa118a00a4/mendable/src/test/java/com/jayasuryat/mendable/MendableReportGeneratorTest.kt#L159) needed fixing.

But, there are several other places where file paths are being constructed with hardcoded file separators, but they all are working fine as they are being used while creating files and `Java`'s `FileSystem` seems to be normalizing those paths while creating those files. Which wasn't the case for the two examples above, as no file was being created there, only the constructed `String` was being used for comparison.
